src/Workspaces/Core/Portable/Workspace/Solution/Checksum.cs

```diff
@@ -189,7 +153,7 @@ public HashData(long data1, long data2, int data3)
                 => x.Equals(y);
 
             public static bool operator !=(HashData x, HashData y)
-                => !x.Equals(y);
+                => !(x == y);
```

Comment:
The original code here was more direct.

Reply:
i prefer this :)   it means the core of hte logic is only ever in one place.  i.e. if we need to change the logic for `==` we don't have to touch `!=`, they're always symettric.